@Yard1 Could you please update the test as well? I got errors in the test:
```
>           output_tokens = prev_output_tokens + new_tokens
E           TypeError: unsupported operand type(s) for +: 'int' and 'list'
```

--- a/tests/engine/test_detokenize.py
+++ b/tests/engine/test_detokenize.py
@@ -42,8 +42,8 @@ def _run_incremental_decode(llama_tokenizer, all_input_ids):
 
 @pytest.mark.parametrize("TRUTH_AND_INPUT_PAIRS", TRUTH_AND_INPUT_PAIRS)
 def test_decode_streaming_english_spaces(llama_tokenizer,
-                                         TRUTH_AND_INPUT_PAIRS):
-    truth, all_input_ids = TRUTH_AND_INPUT_PAIRS
+                                         truth_and_input_pairs):
+    truth, all_input_ids = truth_and_input_pairs
     assert (all_input_ids == llama_tokenizer(
         truth, add_special_tokens=False)["input_ids"])
 

--- a/tests/engine/test_detokenize.py
+++ b/tests/engine/test_detokenize.py
@@ -4,34 +4,38 @@ from transformers import AutoTokenizer
 
 from vllm.transformers_utils.tokenizer import detokenize_incrementally
 
-
-@pytest.fixture(scope="module")
-def llama_tokenizer():
-    return AutoTokenizer.from_pretrained("hf-internal-testing/llama-tokenizer")
-
-
-TRUTH_AND_INPUT_PAIRS = [
-    ("Hello here, this is a simple test",
-     [15043, 1244, 29892, 445, 338, 263, 2560, 1243]),
-    ("<s> vLLM is a high-throughput and memory-efficient inference and serving engine for LLMs. It is designed to be used in production environments, where inference and serving",
-     [
-         1, 325, 2208, 29924, 338, 263, 1880, 29899, 20678, 649, 322, 3370,
-         29899, 8462, 27262, 322, 16330, 6012, 363, 365, 26369, 29879, 29889,
-         739, 338, 8688, 304, 367, 1304, 297, 5802, 23136, 29892, 988, 27262,
-         322, 16330
-     ])
+TRUTH = [
+    "Hello here, this is a simple test",
+    "vLLM is a high-throughput and memory-efficient inference and serving engine for LLMs. It is designed to be used in production environments, where inference and serving",
+    "我很感谢你的热情"
+]
+TOKENIZERS = [
+    "facebook/opt-125m",
+    "gpt2",
+    "bigcode/tiny_starcoder_py",
+    "EleutherAI/gpt-j-6b",
+    "EleutherAI/pythia-70m",
+    "bigscience/bloom-560m",
+    "mosaicml/mpt-7b",
+    "tiiuae/falcon-7b",
+    "meta-llama/Llama-2-7b-hf",
+    "codellama/CodeLlama-7b-hf",
 ]
 
 
-def _run_incremental_decode(llama_tokenizer, all_input_ids):
+def _run_incremental_decode(tokenizer, all_input_ids):
     decoded_text = ""
     offset = 0
     token_offset = 0
     prev_output_tokens = None
     for i in range(len(all_input_ids)):
         new_tokens, text, offset, token_offset = detokenize_incrementally(
-            llama_tokenizer, all_input_ids[:i + 1], prev_output_tokens, offset,
-            token_offset)
+            tokenizer,
+            all_input_ids[:i + 1],
+            prev_output_tokens,
+            offset,
+            token_offset,
+            skip_special_tokens=False)
         decoded_text += text
         if prev_output_tokens is None:
             prev_output_tokens = new_tokens
@@ -40,25 +44,12 @@ def _run_incremental_decode(llama_tokenizer, all_input_ids):
     return decoded_text
 
 
-@pytest.mark.parametrize("TRUTH_AND_INPUT_PAIRS", TRUTH_AND_INPUT_PAIRS)
-def test_decode_streaming_english_spaces(llama_tokenizer,
-                                         truth_and_input_pairs):
-    truth, all_input_ids = truth_and_input_pairs
-    assert (all_input_ids == llama_tokenizer(
-        truth, add_special_tokens=False)["input_ids"])
+@pytest.mark.parametrize("truth", TRUTH)
+@pytest.mark.parametrize("tokenizer_id", TOKENIZERS)
+def test_decode_streaming(tokenizer_id, truth):
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_id)
+    all_input_ids = tokenizer(truth, add_special_tokens=False)["input_ids"]
 
-    decoded_text = _run_incremental_decode(llama_tokenizer, all_input_ids)
-
-    assert decoded_text == truth
-
-
-def test_decode_streaming_chinese_utf8(llama_tokenizer):
-    truth = "我很感谢你的热情"
-    all_input_ids = [
-        30672, 232, 193, 139, 233, 135, 162, 235, 179, 165, 30919, 30210, 234,
-        134, 176, 30993
-    ]
-
-    decoded_text = _run_incremental_decode(llama_tokenizer, all_input_ids)
+    decoded_text = _run_incremental_decode(tokenizer, all_input_ids)
 
     assert decoded_text == truth

--- a/tests/engine/test_detokenize.py
+++ b/tests/engine/test_detokenize.py
@@ -27,10 +27,16 @@ def _run_incremental_decode(llama_tokenizer, all_input_ids):
     decoded_text = ""
     offset = 0
     token_offset = 0
+    prev_output_tokens = None
     for i in range(len(all_input_ids)):
-        text, offset, token_offset = detokenize_incrementally(
-            llama_tokenizer, all_input_ids[:i + 1], offset, token_offset)
+        new_tokens, text, offset, token_offset = detokenize_incrementally(
+            llama_tokenizer, all_input_ids[:i + 1], prev_output_tokens, offset,
+            token_offset)
         decoded_text += text
+        if prev_output_tokens is None:
+            prev_output_tokens = new_tokens
+        else:
+            prev_output_tokens += new_tokens
     return decoded_text
 
 

--- a/vllm/transformers_utils/tokenizer.py
+++ b/vllm/transformers_utils/tokenizer.py
@@ -1,4 +1,4 @@
-from typing import List, Tuple, Union, Optional
+from typing import List, Optional, Tuple, Union
 
 from transformers import (AutoTokenizer, PreTrainedTokenizer,
                           PreTrainedTokenizerFast)
@@ -68,9 +68,10 @@ def get_tokenizer(
 
 
 def _convert_tokens_to_string_with_added_encoders(
-        tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
-        output_tokens: List[str],
-        skip_special_tokens: bool = False):
+    tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
+    output_tokens: List[str],
+    skip_special_tokens: bool,
+) -> str:
     # Adapted from
     # https://github.com/huggingface/transformers/blob/v4.28.0/src/transformers/tokenization_utils.py#L921
     # NOTE(woosuk): The following code is slow because it runs a for loop over
@@ -95,8 +96,8 @@ def _convert_tokens_to_string_with_added_encoders(
     return " ".join(sub_texts)
 
 
-# Based on https://github.com/huggingface/text-generation-inference/\
-# blob/v0.9.4/server/text_generation_server/models/model.py#L62C9-L62C15
+# Based on 
+# https://github.com/huggingface/text-generation-inference/blob/v0.9.4/server/text_generation_server/models/model.py#L62C9-L62C15
 # under Apache 2.0 license
 def detokenize_incrementally(
     tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
@@ -105,7 +106,7 @@ def detokenize_incrementally(
     prefix_offset: int = 0,
     read_offset: int = 0,
     skip_special_tokens: bool = False,
-) -> Tuple[str, str, int, int]:
+) -> Tuple[List[str], str, int, int]:
     new_token_id = all_input_ids[-1]
     # This is the first iteration for this sequence
     if prev_output_tokens is None:

--- a/vllm/transformers_utils/tokenizer.py
+++ b/vllm/transformers_utils/tokenizer.py
@@ -96,7 +96,7 @@ def _convert_tokens_to_string_with_added_encoders(
     return " ".join(sub_texts)
 
 
-# Based on 
+# Based on
 # https://github.com/huggingface/text-generation-inference/blob/v0.9.4/server/text_generation_server/models/model.py#L62C9-L62C15
 # under Apache 2.0 license
 def detokenize_incrementally(

